mobile/library/cc/engine_builder.cc

```diff
@@ -557,13 +556,6 @@ std::unique_ptr<envoy::config::bootstrap::v3::Bootstrap> EngineBuilder::generate
     tag_filter->mutable_typed_config()->PackFrom(tag_config);
   }
 
-  if (!direct_responses_.empty()) {
```

Comment:
having to call ```.addNativeFilter(
              name: "envoy.filters.http.route_cache_reset",
              // swiftlint:disable:next line_length
              typedConfig: "{\"@type\":\"type.googleapis.com/envoymobile.extensions.filters.http.route_cache_reset.RouteCacheReset\"}")```

in every test that uses `DirectResponse` is not great. In fact, I think that this may break the `DirectResponse` for outside-of-Envoy-repo users. 

So the EM binary that we create  does not expose `register_test_extensions` function or anything like that. Now, some of EM customers have tests (in Lyft repo for example) that depend on `DirectResponse` API. These tests would need to be updated to add `envoy.filters.http.route_cache_reset` native filter but such addition would result in a crash as the corresponding route cache reset proto is not registered as part of the release EM binary anymore.

Trying to think of a way to solve this (without removing `DirectResponse` API altogether) but nothing obvious come to my mind - a bit cumbersome thing would be to add another build option to specify whether DirectResponse API should be supported or not. Feels a little bit heavy.

Reply:
we can register `envoy.filters.http.route_cache_reset` for builds that we use at Lyft as a hacky workaround (with a note that any other customer would need to do the same if they wanted to use DirectResponse API)

Reply:
isn't direct response a test only thing we're trying to get rid of?

Reply:
Yes, that's the plan realistically speaking I am not going to get to removing it from Lyft codebase until after I am done with an initial version of iOS proxy support which I hope to be done with next week. I need to find out a replacement for EM direct response too.

Anyway, a stop gap solution could be us (lyft) registering required extensions on our side. 

Last thing that I am thinking about is to whether we should override a constructor of `TestEngineBuilder.swift` so that it does call `.addNativeFilter( name: "envoy.filters.http.route_cache_reset", // swiftlint:disable:next line_length typedConfig: "{\"@type\":\"type.googleapis.com/envoymobile.extensions.filters.http.route_cache_reset.RouteCacheReset\"}")` for us to avoid multiple changes downstream. Any thoughts?

Reply:
From the perspective of Envoy Mobile at Lyft the most short term gap solution (until we do iOS proxies and finish the rollout) would be to continue to maintain the `DirectResponse` API in its current form. We could work on ripping out/changing our integration tests so that they do not rely on `DirectResponse` API but that I think would take us significantly more time than just continue to support this API for the time being. At the same time it would delay our rollout which we consider to be of a significantly higher priority for Lyft (and I would argue that it helps the project as a whole too).

If you do not have a strong opinion against it I think that we should add a build option such as `direct_response_api` enabled (or something along these lines) that would compile in/compile out the support for `DirectResponse` API conditionally. I realize that it's not ideal.

I am happy to make the build/compilation flag change to not make you do work just to make Lyft happy. I could work on that tmrw. Let me know what you think.

Reply:
I've spent some time too and I think I have a third option that might be better than the two presented.

if I set the type url without using the proto, I think we can avoid the engine builder needing the route cache reset filter.  That'd change the import pain from adding a filter, to making sure to pull in test extensions in the build file (which most tests already do).  Testing CI now (unfortunately still can't build swift locally) but this strategy worked when I tested it with the socket tagging filter so I'm optimistic it'll pass once I fix any typos.

Reply:
Also now that I think of it, if this new version works you could avoid any build pain if you use the route cache filter in your own (custom) internal list of builders